### PR TITLE
Sviatoslav branch

### PR DESCRIPTION
**Pull request by CODER591**
Fixed logger in conversions , fixed writing to file,
Added SendPeningRequests back to client  and other minor improvements

### CODE REVIEW

File: Server/Logger/logger.cpp [hunk 1]
```diff
@@ -1,8 +1,10 @@
 #pragma once
```
Comment (KMarkiyan) [inline Server/Logger/logger.cpp]: Change #pragma once with #ifndef...
http://web.archive.org/web/20160430022340/http://google.github.io/styleguide/cppguide.html#The__define_Guard

File: Server/Logger/logger.cpp [hunk 2]
```diff
@@ -15,44 +17,54 @@ QString Logger::ConvertQuint8ToString(quint8 & num)
...
-	QString for_empty = ConvertQuint8ToString(type) +")" + txt + "+ empty{}";
-	QString for_struct;
+   QString outingString;  // date (d/m/y) time (h/m/s)
+	 QString txt = QDateTime::currentDateTime().toString("dd:MM:yyyy hh:mm:ss ");
+	 quint8 type = Parser::getRequestType(raw_data);
+	 QString for_empty = ConvertQuint8ToString(type) +")" + txt + "+ empty{}";
+	 QString for_struct;
     switch (type) {
       case (quint8)ServerRequests::LOGIN_SUCCEED: {
 		outingString = "Server output: LOGIN_SUCCEED ( " + for_empty;
```
Comment (KMarkiyan) [inline Server/Logger/logger.cpp]: Different tabs.
http://web.archive.org/web/20160430022340/http://google.github.io/styleguide/cppguide.html#Spaces_vs._Tabs

File: Server/Logger/logger.h [hunk 3]
```diff
@@ -12,8 +12,12 @@ class Logger {
    static void LogOut(QByteArray);
    static QString Log_User(Client &cl);
  private:
-   static void WriteLogToFile(QString&,QString&);
-   static QString ConvertQuint8ToString(quint8& num);   
+   //static void WriteLogToFile(QString&,QString&);
```
Comment (KMarkiyan) [inline Server/Logger/logger.h]: Remove this comment

File: Server/RequestProcessing/add_friend_request.cpp [hunk 4]
```diff
@@ -62,6 +62,8 @@ bool AddFriendRequest::SendResponde() {
     client_socket_->waitForBytesWritten(3000);
     client_socket_->disconnectFromHost();
   }
-
-  return false;
+  
+  
+  
+    return false;
```
Comment (KMarkiyan) [inline Server/RequestProcessing/add_friend_request.cpp]: remove empt lines

File: Server/RequestProcessing/LoginRequest.cpp [hunk 5]
```diff
@@ -50,40 +48,80 @@ bool LoginRequest::SendResponde() {
...
+    output_socket.disconnectFromHost();
+  }
+
+  return false;
+}
+
+void LoginRequest::SendingPendingRequests(QTcpSocket &notifiying_requester) {
+  
+  // this function should send to requester pending requests and pending notifications 
+ // QTcpSocket notifiying_requester;
```
Comment (FedorenkoAndriy) [inline Server/RequestProcessing/LoginRequest.cpp]: remove

File: Server/RequestProcessing/LoginRequest.cpp [hunk 6]
```diff
@@ -50,40 +48,80 @@ bool LoginRequest::SendResponde() {
...
+
   } else {
     //sending ServerRequests::LOGIN_FAILED
     QByteArray b = Parser::Empty_ToByteArray(response_to_requester_);
     b.append(Parser::GetUnpossibleSequence());
     Logger::LogOut(b);
     client_socket_->write(b);
     client_socket_->waitForBytesWritten(3000);
-    client_socket_->disconnectFromHost();
+    //client_socket_->disconnectFromHost();
```
Comment (FedorenkoAndriy) [inline Server/RequestProcessing/LoginRequest.cpp]: remove

**Pull request merged**